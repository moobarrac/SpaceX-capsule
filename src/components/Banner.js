import React, { useState } from 'react';
import Space from "../assests/images/spacex.jpg";

const Banner = () => {
  const [scale, setScale] = useState(1);
  return (
    <div
      className="bg-cover bg-no-repeat bg-center h-80" 
      style={{ backgroundImage: `url(${Space})` }}
      onMouseEnter={() => setScale(1.20)}
      onMouseLeave={() => setScale(1)}
    >
      <div className="px-3 py-2 bg-black bg-opacity-80 text-white text-center rounded-lg animation-fade-in-up">
        <h1 className="text-white text-4xl font-medium" style={{ transform: `scale(${scale})`}}>SpaceX capsules</h1>
        <p className="text-sm">Explore the latest space missions!</p>
      </div>
    </div>
  )
}

export default Banner