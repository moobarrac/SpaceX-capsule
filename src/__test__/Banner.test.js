/** @jest-environment jsdom */
import React from 'react';
import { render, container } from '@testing-library/react';
import '@testing-library/jest-dom';
import Banner from '../components/Banner';
jest.mock('../assests/images/spacex.jpg', () => 'mocked-spacex-url');

describe('Banner component', () => {

  it('renders the correct text in the h1 tag', () => {
    render(<Banner />);
    expect(container).toMatchSnapshot();
  });
});

