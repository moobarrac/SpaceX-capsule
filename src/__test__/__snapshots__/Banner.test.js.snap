// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component renders the correct text in the h1 tag 1`] = `undefined`;
